### PR TITLE
OutsideExecution (SNIP-9)

### LINKED ISSUE #1110: Implementation of SNIP-9
**🧐 Motivation**
SNIP-9 Outside Execution (https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-9.md) is implemented in ArgentX & Braavos account contracts.

**📝 Details**
Do you plan to implement soon this very useful feature?

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #1110  

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Added entry to CHANGELOG.md 
- [ ] Tried the feature on a public network

### CODE REVIEW

**Review by andrew-fleming — COMMENTED** [review]
The PR looks really good, Eric! I left some minor comments and questions

**Review by immrsd — COMMENTED** [review]
Well done, Eric! Left a couple of suggestions

File: docs/modules/ROOT/pages/api/account.adoc [hunk 1]
```diff
@@ -566,6 +567,132 @@ Emitted when a `public_key` is added.
...
+==== Embeddable functions
+
+[.contract-item]
+[[OutsideExecutionComponent-execute_from_outside_v2]]
+==== `[.contract-item-name]#++execute_from_outside_v2++#++(ref self: ContractState, outside_execution: OutsideExecution, signature: Span<felt252>) → Array<Span<felt252>>++` [.item-kind]#external#
+
+Allows anyone to submit a transaction on behalf of the account as long as they
+have the relevant signatures.
+
+This method allows reentrancy. A call to `__execute__` or `execute_from_outside` can
```
[the remark this answers is not in the document]
Comment (ericnordelo) [reply docs/modules/ROOT/pages/api/account.adoc:663]: Done!

File: packages/account/src/tests/mocks/src9_mocks.cairo [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    component!(path: AccountComponent, storage: account, event: AccountEvent);
+    component!(path: SRC5Component, storage: src5, event: SRC5Event);
+    component!(path: OutsideExecutionComponent, storage: src9, event: SRC9Event);
+
+    // Account
+    #[abi(embed_v0)]
+    impl SRC6Impl = AccountComponent::SRC6Impl<ContractState>;
+    #[abi(embed_v0)]
+    impl PublicKeyImpl = AccountComponent::PublicKeyImpl<ContractState>;
```
[the remark this answers is not in the document]
Comment (ericnordelo) [reply packages/account/src/tests/mocks/src9_mocks.cairo:16]: You are right, but I wanted to let the mock closer to a preset reference for users, as it is the only guide they currently have until we add this to Wizard. The intention is not to make mocks references at all, but in the meantime I think they can be helpful.

**Comment by immrsd** [comment]
What do you think about adding SRC-9 to account presets?

**Comment by ericnordelo** [comment]
> What do you think about adding SRC-9 to account presets?

I think we should add it to wizard, not to presets. In general, presets are a temporary solution to examples out of wizard, but we should always favor wizard, and we should avoid a situation where we start adding many presets, even when some of them are useful.

**Review by andrew-fleming — APPROVED** [review]
LGTM!

**Pull request merged**